public/js/controllers/main.controller.js

```diff
@@ -20,25 +21,61 @@ angular.module('main.controller', [])
                         $scope.newTodo = {};
                         $scope.loading = false;
                     })
-                    .error(function (data, status) { $scope.loading = false; });
+                    //TODO: it would be nice if we can bind error status right to the view
+                    .error(function(data, status) {
+                        $scope.loading = false;
+                        $scope.error = status;
+                    });
             }
         };
 
-        $scope.deleteTodo = function (id) {
+        $scope.removeItemFromArray = function (array, item) {
+            var index = 0;
+            array.some(function (entry, i) {
+                            if (entry.id === item.id) {
+                                index = i;
+                                return true;
+                            }
+            });
+
+            array.splice(index, 1);
+        }
+
+        $scope.replaceItem = function (sourceArray, destArray, item) {
```

Comment:
It seems that method replace item to another in one array... And method can be private too